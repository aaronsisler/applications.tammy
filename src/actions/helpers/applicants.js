export const setApplicants = (applicants) => ({
    type: 'SET_APPLICANTS',
    applicants
});
