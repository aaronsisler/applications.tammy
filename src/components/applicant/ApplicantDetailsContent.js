import React from 'react';
import PropTypes from 'prop-types';

export default class ApplicantDetailsContent extends React.Component {
    constructor(props) {
        super(props);
    }

    render() {
        return (
            <div className="applicant_details_content">
                Applicant Content goes here
            </div>
        );
    }
}

ApplicantDetailsContent.propTypes = {
    applicant: PropTypes.object.isRequired,
};

