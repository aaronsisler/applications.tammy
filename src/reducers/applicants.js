const applicantsReducerDefaultState = [];

const applicantsReducer = (state = applicantsReducerDefaultState, action) => {
    switch (action.type) {
        case 'SET_APPLICANTS':
            return action.applicants || applicantsReducerDefaultState;
        default:
            return state;
    }
};

export default applicantsReducer;

