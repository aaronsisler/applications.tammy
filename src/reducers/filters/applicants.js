const applicantsFilterReducerDefaultState =
{
    text: '',
};

const applicantsFilterReducer = (state = applicantsFilterReducerDefaultState, action) => {
    switch (action.type) {
        case 'SET_APPLICANTS_TEXT_FILTER':
            return {
                ...state,
                text: action.text
            };
        default:
            return state;
    }
};

export default applicantsFilterReducer;
