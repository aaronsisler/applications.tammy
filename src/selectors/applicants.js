// Get filtered applicants
export default (applicants = [], { text = '' }) => applicants
    .filter((applicant) => {
        const lastNameMatch = applicant.user.lastName.toLowerCase().includes(text.toLowerCase());
        const firstNameMatch = applicant.user.firstName.toLowerCase().includes(text.toLowerCase());
        return lastNameMatch || firstNameMatch;
    })
