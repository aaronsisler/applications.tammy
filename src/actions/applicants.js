import database from 'Firebase/firebase';
import { setApplicants } from 'Actions/helpers/applicants';
import { setApplicantsTextFilter } from 'Actions/filters/applicantsText';

export const startSetApplicants = () => (dispatch, getState) => {
    const { positionId } = getState().workflow.position;

    return database.ref(`applicants/${positionId}`).once('value').then((snapshot) => {
        const applicants = [];
        snapshot.forEach((childSnapshot) => {
            const applicant = childSnapshot.val();
            const applicantNotes = applicant.applicantNotes
                ? [...Object.values(applicant.applicantNotes)]
                : [];
            applicants.push({
                applicantId: childSnapshot.key,
                positionId,
                ...childSnapshot.val(),
                applicantNotes,
            });
        });

        return dispatch(setApplicants(applicants));
    });
}

export const startSetApplicantsTextFilter = (text) => (dispatch) => dispatch(setApplicantsTextFilter(text));
