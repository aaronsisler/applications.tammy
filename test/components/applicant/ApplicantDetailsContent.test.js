import React from 'react';
import { shallow } from 'enzyme';
import ApplicantDetailsContent from 'Applicant/ApplicantDetailsContent';
import { applicant } from '../../fixtures/applicants';

describe('ApplicantDetailsContent', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(
            <ApplicantDetailsContent
                applicant={applicant}
            />
        );
    })

    it('should render ApplicantDetailsContent correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });
});
