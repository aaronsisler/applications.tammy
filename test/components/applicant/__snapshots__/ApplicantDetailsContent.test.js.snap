// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplicantDetailsContent should render ApplicantDetailsContent correctly 1`] = `
<div
  className="applicant_details_content"
>
  <Tabs
    defaultFocus={false}
    defaultIndex={null}
    forceRenderTabPanel={false}
    selectedIndex={null}
  >
    <TabList
      className="react-tabs__tab-list"
    >
      <Tab
        className="react-tabs__tab"
        disabledClassName="react-tabs__tab--disabled"
        focus={false}
        id={null}
        panelId={null}
        selected={false}
        selectedClassName="react-tabs__tab--selected"
      >
        Workflow
      </Tab>
      <Tab
        className="react-tabs__tab"
        disabledClassName="react-tabs__tab--disabled"
        focus={false}
        id={null}
        panelId={null}
        selected={false}
        selectedClassName="react-tabs__tab--selected"
      >
        Documents
      </Tab>
      <Tab
        className="react-tabs__tab"
        disabledClassName="react-tabs__tab--disabled"
        focus={false}
        id={null}
        panelId={null}
        selected={false}
        selectedClassName="react-tabs__tab--selected"
      >
        Info
      </Tab>
    </TabList>
    <TabPanel
      className="react-tabs__tab-panel"
      forceRender={false}
      selectedClassName="react-tabs__tab-panel--selected"
    >
      <Connect(ApplicantProcessContainer)
        applicant={
          Object {
            "applicantId": "firstApplicantId",
            "applicantNotes": Array [
              Object {
                "noteMessage": "First note",
              },
              Object {
                "noteMessage": "Second note",
              },
            ],
            "applicantStatus": "APPLIED",
            "positionId": "firstPositionId",
            "user": Object {
              "addressLine1": "123 Evergreen Terrace",
              "addressLine2": "Apt. 456",
              "city": "Durham",
              "displayName": "Johnny Appleseed",
              "displayPhoneNumber": "919-123-4567 ext. 12345",
              "email": "johnny.appleseed@gmail.com",
              "firstName": "Johnny",
              "lastName": "Appleseed",
              "middleName": "Middleton",
              "phoneAreaCode": "919",
              "phoneExt": "12345",
              "phoneLineNumber": "4567",
              "phonePrefix": "123",
              "postalCode": "27540",
              "state": "NC",
            },
            "userDocuments": Array [
              Object {
                "dateUploaded": "2018/01/01 01:23:45A",
                "documentName": "Document Name 1",
                "downloadUrl": "downloadUrl1",
                "userDocumentId": "firstDocumentId",
              },
              Object {
                "dateUploaded": "2018/02/02 01:23:45A",
                "documentName": "Document Name 2",
                "downloadUrl": "downloadUrl2",
                "userDocumentId": "secondDocumentId",
              },
            ],
          }
        }
      />
    </TabPanel>
    <TabPanel
      className="react-tabs__tab-panel"
      forceRender={false}
      selectedClassName="react-tabs__tab-panel--selected"
    >
      <ApplicantUserDocuments
        userDocuments={
          Array [
            Object {
              "dateUploaded": "2018/01/01 01:23:45A",
              "documentName": "Document Name 1",
              "downloadUrl": "downloadUrl1",
              "userDocumentId": "firstDocumentId",
            },
            Object {
              "dateUploaded": "2018/02/02 01:23:45A",
              "documentName": "Document Name 2",
              "downloadUrl": "downloadUrl2",
              "userDocumentId": "secondDocumentId",
            },
          ]
        }
      />
    </TabPanel>
    <TabPanel
      className="react-tabs__tab-panel"
      forceRender={false}
      selectedClassName="react-tabs__tab-panel--selected"
    >
      <ApplicantUserInfo
        user={
          Object {
            "addressLine1": "123 Evergreen Terrace",
            "addressLine2": "Apt. 456",
            "city": "Durham",
            "displayName": "Johnny Appleseed",
            "displayPhoneNumber": "919-123-4567 ext. 12345",
            "email": "johnny.appleseed@gmail.com",
            "firstName": "Johnny",
            "lastName": "Appleseed",
            "middleName": "Middleton",
            "phoneAreaCode": "919",
            "phoneExt": "12345",
            "phoneLineNumber": "4567",
            "phonePrefix": "123",
            "postalCode": "27540",
            "state": "NC",
          }
        }
      />
    </TabPanel>
  </Tabs>
</div>
`;
